### PR TITLE
[plugin-kafka] Fix flaky kafka tests

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus-tests/src/main/resources/story/system/Kafka.story [hunk 1]
```diff
@@ -24,7 +24,7 @@ When I execute steps with delay `PT1S` at most 30 times while variable `messageC
 |When I peek consumed `vividus` Kafka messages to scenario variable `messages`                                                       |
 |When I save number of elements from `${messages}` found by JSON path `$..[?(@.key == "failed")]` to scenario variable `messageCount`|
 When I drain consumed `vividus` Kafka messages to scenario variable `consumed-messages`
-Then `${consumed-messages}` is equal to `[{"key" : "failed"}, {"key" : "passed"}]`
+Then JSON element from `${consumed-messages}` by JSON path `$` is equal to `[{"key" : "failed"}, {"key" : "passed"}]`IGNORING_ARRAY_ORDER
```
Comment (valfirst) [inline vividus-tests/src/main/resources/story/system/Kafka.story:27]:
```suggestion
Then JSON element from `${consumed-messages}` by JSON path `$` is equal to `[{"key" : "failed"}, {"key" : "passed"}]` ignoring array order
```

File: vividus-tests/src/main/resources/story/system/Kafka.story [hunk 2]
```diff
@@ -55,5 +55,5 @@ When I execute steps with delay `PT1S` at most 30 times while variable `eventCou
 |When I peek consumed `vividus` Kafka events to scenario variable `events`                                                          |
 |When I save number of elements from `${events}` found by JSON path `$..[?(@.status == "failed")]` to scenario variable `eventCount`|
 When I drain consumed `vividus` Kafka events to scenario variable `consumed-events`
-Then `${consumed-events}` is equal to `[{"status" : "failed"}, {"status" : "passed"}]`
+Then JSON element from `${consumed-events}` by JSON path `$` is equal to `[{"status" : "failed"}, {"status" : "passed"}]`IGNORING_ARRAY_ORDER
```
Comment (valfirst) [inline vividus-tests/src/main/resources/story/system/Kafka.story:58]:
```suggestion
Then JSON element from `${consumed-events}` by JSON path `$` is equal to `[{"status" : "failed"}, {"status" : "passed"}]` ignoring array order
```

File: vividus-plugin-kafka/src/test/java/org/vividus/steps/kafka/KafkaStepsIntegrationTests.java [hunk 3]
```diff
@@ -198,8 +203,20 @@ void shouldProduceEventWithKey() throws InterruptedException, ExecutionException
...
+        List<LoggingEvent> events = logger.getLoggingEvents();
+        assertThat(events, hasSize(3));
+        assertEquals(info(LISTENER_STARTED), events.get(0));
+        assertEquals(info(LISTENER_STOPPED), events.get(1));
+        LoggingEvent keysLogEvent = events.get(2);
+        assertEquals(Level.INFO, keysLogEvent.getLevel());
+        assertEquals("Saving events with the keys: {}", keysLogEvent.getMessage());
+        List<Object> arguments = keysLogEvent.getArguments();
+        assertThat(arguments, hasSize(1));
+        List<String> keys = Stream.of(arguments.get(0).toString().split(",")).map(String::strip).sorted()
```
Comment (ikalinin1) [inline vividus-plugin-kafka/src/test/java/org/vividus/steps/kafka/KafkaStepsIntegrationTests.java:215]: any reasons to sort ?

Comment (uarlouski) [reply vividus-plugin-kafka/src/test/java/org/vividus/steps/kafka/KafkaStepsIntegrationTests.java:215]: different order of events

Comment (ikalinin1) [reply vividus-plugin-kafka/src/test/java/org/vividus/steps/kafka/KafkaStepsIntegrationTests.java:215]: I mean there are matchers for like hasItesm containsInAnyOrder etc

Comment (uarlouski) [reply vividus-plugin-kafka/src/test/java/org/vividus/steps/kafka/KafkaStepsIntegrationTests.java:215]: these matchers are good if we have distinct values, in out case we have duplicates

**Review by valfirst — APPROVED** [review]

**Pull request merged**